File: cpp/se/ArDomainObjectType.cpp
#include <string>

#include "ArDomainObjectType.hpp"

const std::string & ArDomainObjectType::getName() const {
    static const std::string name = "ardom";
    return name;
}

const std::string & ArDomainObjectType::getURI() const {
    static const std::string uri = "urn:X-ar:params:xml:ns:ardomain-1.0";
    return uri;
}

const std::string & ArDomainObjectType::getSchemaLocation() const {
    static const std::string schemaLocation =
            "urn:X-ar:params:xml:ns:ardomain-1.0 ardomain-1.0.xsd";
    return schemaLocation;
}

const std::string & ArDomainObjectType::getIdentType() const {
    static const std::string ident = "name";
    return ident;
}

File: src/main/java/com/ausregistry/jtoolkit2/se/ArDomainObjectType.java
package com.ausregistry.jtoolkit2.se;

/**
 * Represents the ardomain-1.0 EPP extension object.
 * @deprecated Should instead use {@link ExtendedObjectType#AR_DOMAIN}.
 */
class ArDomainObjectType implements ObjectType {
    private static final long serialVersionUID = 992842005698675196L;

    public String getName() {
        return "ardom";
    }

    public String getURI() {
        return "urn:X-ar:params:xml:ns:ardomain-1.0";
    }

    public String getSchemaLocation() {
        return "urn:X-ar:params:xml:ns:ardomain-1.0 ardomain-1.0.xsd";
    }

    public String getIdentType() {
        return "name";
    }
}

